### DISCUSSION TITLE
Should selecting a Bank open it so that I can select a Voice?

### DISCUSSION THREAD

**Discussion by Onefelloff** [probonopd/MiniDexed#425] [Q&A] [answered]
Hello, I'm new to GitHub, to the whole DIY Synth thing and to programming (well other than a lot of Basic in the 80's) I am however a big DX7 fan and this project is great. I have built up a working version (Pi4, Rasipaudio, screen and rotary encoder, there will be 5 pin midi) which sounds great. but I'm not sure that the behaviour I'm experiencing with Banks and Voices is normal or not.

I've set up a sysex/voice folder structure and copied our some sysex files which seem to have the correct filename structure and I can access all of the voices from the "Voice" menu. The bank menu seems just to be a list of the bank names (only two at the moment, but there were more) that don't seem to do anything when selected. I was expecting the bank to open so that I can select a voice from within it.

Is this standard behaviour?

Sorry if this is a duplicate question, I have had a look around and cant find anything quite the same.

Could anyone advise or tell me what I've done wrong?

I've uploaded the config I'm using [here](https://github.com/Onefelloff/miniDexed)

Many thanks in advance

<img width="742" alt="Screenshot 2023-01-22 at 21 26 26" src="https://user-images.githubusercontent.com/123325746/213941289-3548018c-9034-4c80-9fdb-6787ecbbf6f6.png">

**Comment by diyelectromusic** [comment, marked as answer]
At present, changing the bank does not automatically change the voice from that bank as well, so it is really selecting the next bank to use for the next time you then go to the voice change menu. I'm not sure what is the most useful behaviour at the moment.

One update that has recently gone in is to auto-bank-switch when you scroll up past voice 32 or down past 1 on the current bank.  This allows quick selection through the voices across multiple banks.

The "jury is still out" on how banks and programs should be interacting, especially over MIDI too... there are a few open issues/PRs where some of the ideas are being discussed: see https://github.com/probonopd/MiniDexed/pull/395 or https://github.com/probonopd/MiniDexed/issues/391 and this exact issue was also raised here: https://github.com/probonopd/MiniDexed/pull/392

Kevin

**Reply by probonopd** [reply]
> At present, changing the bank does not automatically change the voice from that bank as well, so it is really selecting the next bank to use for the next time you then go to the voice change menu. I'm not sure what is the most useful behaviour at the moment.

What do Yamaha instruments from the 80s do in this regard?

**Reply by Onefelloff** [reply]
Hi, thanks for the reply, so it sounds like what I’m experiencing is exactly what is supposed to be happening.  I’d read #392 and understood the scrolling through the voices thing, which does work well as long as you’ve not got too many banks loaded. 

Selecting a bank (internal or external) and then a voice is exactly what the DX7 does, but you’ve only got 32 buttons to choose from to select a voice or 64 on a DX7II

I guess if the display could show the bank (even if it was only the bank number) when scrolling through the voices it would be easier to see where you are in the list. It’s easy enough to manage as it is I’ll just be a bit selective about how many banks I load.

it sounds great

Many thanks

**Reply by diyelectromusic** [reply]
> What do Yamaha instruments from the 80s do in this regard?

My DX11 will switch bank to the same numbered voice.  So if its is on "Internal A/Voice 3" and you swtich to "Internal B" it will change to "Internal B/Voice 3".

I've never owned a DX7 :)

**Reply by Onefelloff** [reply]
Hi

I've been giving a bit of thought to the Bank / Voice thing and two options present themselves to me to aid with regard to navigation and voice selection.

1. Keep operation exactly as it is but add a bank number to the voice display somehow, just so its easy to track where in the voice list you are, though I guess the bank number would have to be three digits which I guess there isn't room for, is there room for 2 digits?  100 banks, 3200 voices seems like  enough of a selection to me, my modern synths seem to manage with 1024.

2. Once you've selected a bank and then move back to select a voice you are taken to voice 01 of that bank, it's then clear you are in a different place from where you were before.

I'm not sure how easy these things are to do, I certainly wouldn't know where to start and once you get your head around it as it is fine. Its just that once I start scrolling I get a bit lost.

Just so you know I'm very grateful for the functionality MiniDexed currently has and expect you've already thought of these things.

**Comment by Onefelloff** [comment]
Hi, Ive got a Mk1 from 1984, the same one from new, sadly I am that old!

Mark

**Reply by probonopd** [reply]
So you can confirm that the currently implemented behavior of MiniDexed is the same as the DX7 Mk1 from 1984?
Then I'd say for the sake of compatibility, we should keep it so? (I am thinking of all the MIDI gadgets that are used to "drive" DXs that MiniDexed wants to stay compatible with.)

**Reply by Onefelloff** [reply]
I think so, except of course that you only get to select from 32 voices once you've selected a bank or 64 on a DX7II.
You definitely can't scroll though all of the patches without changing the bank.

Mark

**Reply by diyelectromusic** [reply]
With these things I always go with "least surprise" operation if possible.  For me least surprise is mirroring what the DX11 does, so when on voice 4 bank A, swapping to bank B should select bank B voice 4 automatically.  If the DX7 does that too, I'll add that in, but it would be nice if that could be confirmed!

I don't think that should happen via MIDI though, as I would expect a DAW/etc to send bank sel/program change under its control, and probably as a set of messages...

**Comment by Onefelloff** [comment]
Hi, I thought I'd upload a short video of bank / voice selection. as you can see the Voice Number and name is prefixed by where it's from. I thought this might be better than trying to describe it.

Sorry about my fat fingers.

https://user-images.githubusercontent.com/123325746/215124504-a9f82d91-5453-4fb3-bb94-8eca959484d1.mov
